docs/ingress-gateway-external-lb/ingress-gateway.yaml

```diff
@@ -0,0 +1,124 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: asm-ingressgateway
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: asm-ingressgateway
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: asm-ingressgateway
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: asm-ingressgateway
+subjects:
+- kind: ServiceAccount
+  name: asm-ingressgateway
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: asm-ingressgateway
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: asm-ingressgateway
+      istio: ingressgateway
```

Comment:
Some existing (not all) samples like the Edge to Mesh guide have the label `asm: ingressgateway` for the `Service` and `Deployment`, do you want to change `istio:` by `asm:` too if that makes sense in the context of your doc?

Reply:
I actually switched it back to `istio:` because it seems like most examples use that is what they use in the docs:
https://cloud.google.com/service-mesh/docs/gateways#in-cluster

Reply:
Yep, not everywhere. Background of my comment is that it's what we did 1+ year ago in here https://cloud.google.com/architecture/exposing-service-mesh-apps-through-gke-ingress based on Ameer feedback. But yeah, I see/know that's not everywhere.

Reply:
Yep, I was part of those discussions as well. I tend to use Bank of Anthos for as my "sample apps" and it still uses `istio: ingressgateway`.

Reply:
I can bring this up with the ASM docs team and see if there are any plans to change to the official docs. The doc this will be associated with will be published alongside those docs.

Reply:
Aaron, Christine, the TWs, and I just talked about this `istio: ingressgateway` vs `asm: ingressgateway` thread. 
We concluded that we should have a discussion around standardizing these things (perhaps in Christine's Google-internal [go/asm-ingress-docs](http://go/asm-ingress-docs) doc) across the repo and cloug.google.com tutorials.

But I think, for this pull-request, it shouldn't be a blocker.

If I had to vote, I'd choose: 
1. Use`istio:` for now.
2. TWs/DPEs have discussion about consistent naming. Settle on a naming convention.
3. DPEs make code changes in this repo to update samples to use the chosen naming convention.

Reply:
Thanks, it is currently using `istio: ingressgateway`